pkg/kubeapiserver/options/authentication.go

```diff
@@ -406,29 +412,24 @@ func (s *BuiltInAuthenticationOptions) ToAuthenticationConfig() (kubeauthenticat
 	return ret, nil
 }
 
-func (o *BuiltInAuthenticationOptions) ApplyTo(c *genericapiserver.Config) error {
+func (o *BuiltInAuthenticationOptions) ApplyTo(c *genericapiserver.Config, extclient kubernetes.Interface, versionedInformer informers.SharedInformerFactory) error {
```

Comment:
Add a comment indicating the order dependencies.  Because this is *only* for the kube-apiserver, I can accept those deps for now as we unwind this, but I need to see the godoc for it the preflight checks here too.  things like "is OpenAPIConfig set"

Reply:
I added godoc and the prefight check for OpenAPIConfig, but the egressSelector seems optional so I don't think that method can enforce it